u8timeseries/models/RNN_model.py

```diff
@@ -0,0 +1,469 @@
+
+    def fit(self,
+            series: TimeSeries,
+            val_series: Optional[TimeSeries] = None,
+            verbose: bool = False) -> None:
+        """
+        :param series: The training time series
+        :param val_series: Optionally, a validation time series that will
+                           be used to compute validation loss throughout training
+        """
+
+        super().fit(series)
+
+        if self.from_scratch:
+            shutil.rmtree(_get_checkpoint_folder(self.work_dir, self.model_name), ignore_errors=True)
+
+        if self.batch_size is None:
+            self.batch_size = len(series) // 10
+            print('No batch size set. Using: {}'.format(self.batch_size))
+
+        # Prepare training data:
+        dataset = TimeSeriesDataset1D(series, self.seq_len, self.output_length)
+        train_loader = DataLoader(dataset, batch_size=self.batch_size, shuffle=True,
+                                  num_workers=0, pin_memory=True, drop_last=True)
+        if len(train_loader) == 0:
+            raise ValueError('The provided training time series is too short for obtaining even one training point.')
+
+        # Prepare validation data:
+        if val_series is not None:
+            val_dataset = TimeSeriesDataset1D(val_series, self.seq_len, self.output_length)
+            val_loader = DataLoader(val_dataset, batch_size=self.batch_size, shuffle=False,
+                                    num_workers=0, pin_memory=True, drop_last=False)
+            if len(val_dataset) == 0 or len(val_loader) == 0:
+                raise ValueError('The provided validation time series is too short for this model output length.')
+        else:
+            val_loader = None
+
+        # Tensorboard
+        runs_folder = _get_runs_folder(self.work_dir, self.model_name)
+        if self.log_tensorboard:
+            if self.from_scratch:
+                shutil.rmtree(runs_folder, ignore_errors=True)
+                tb_writer = SummaryWriter(runs_folder)
+                dummy_input = torch.empty(self.batch_size, self.seq_len, self.input_size).to(self.device)
+                tb_writer.add_graph(self.model, dummy_input)
+            else:
+                tb_writer = SummaryWriter(runs_folder, purge_step=self.start_epoch)
+        else:
+            tb_writer = None
+
+        self._train(train_loader, val_loader, tb_writer, verbose)
+
+        if tb_writer is not None:
+            tb_writer.flush()
+            tb_writer.close()
+
+    def predict(self, n: int) -> TimeSeries:
+        """
+        :return: A TimeSeries containing the `n` next points, starting after the end of the training time series.
+        """
+
+        super().predict(n)
+
+        scaled_series = self.training_series.values()[-self.seq_len:]
+        pred_in = torch.from_numpy(scaled_series).float().view(1, -1, 1).to(self.device)
+        test_out = []
+        self.model.eval()
+        for i in range(n):
```

Comment:
Apologies in advance if I'm just misunderstanding the code. But it seems to me that we're only ever using the first element of the predictions by the RNN. If that's the case, when does it make sense to set output_length to a value larger than 1?

Reply:
That is a good question. It's not used for predictions, but for training. At training time, we can compute the loss over a predicted sequence (see line 329). Note that this means that when output_length > 1, we do not train for the same task as what is done at prediction time. We might want to have this behaviour captured by a hyper-parameter at some point.